app/components/UI/DatePicker/DatePicker.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+
+import {
+  isSameMonth, subMonths, subDays,
+  addMonths, getMonth, getYear,
+} from 'date-fns';
+import { ClassNames, DayPicker } from 'react-day-picker';
+import 'react-day-picker/dist/style.css';
+import styles from 'react-day-picker/dist/style.module.css';
+
+type Props = {
+  selectedDate: Date,
+  setSelectedDate: (selectedDate: Date) => void,
+  hidePastDays: boolean,
+  monthsBackToBeShown: number,
+  monthsForwardToBeShown: number,
+  daysToDisable: Date[],
+};
+
+export default function DatePicker({
+  setSelectedDate,
+  selectedDate,
+  hidePastDays = false,
+  monthsBackToBeShown = 12,
+  monthsForwardToBeShown = 3,
+  daysToDisable = [],
+}: Props) {
+  const today = new Date();
+  const initialShownDate = hidePastDays ? today : subMonths(today, monthsBackToBeShown);
+  const finalShownDate = addMonths(today, monthsForwardToBeShown);
+  const disabledDays = [...daysToDisable];
```

Comment:
Please add the next type to `disableDays ` to avoid warnings from TypeScript:
```suggestion
  const disabledDays : Matcher[] = [...daysToDisable];
```
The origin of this change is because the `disabled` prop from DayPicker expects to receive something of Type `Matcher[]`. Matcher could be a Date or a `DateRange` ({from: ... to: ...}) As you assign a DateRange to disabledDays [here](https://github.com/EzeLamar/sports-booking/pull/12/files#diff-20c40455d2fbc8cf970912a92db536e66d5648300bb9e2135f394483779276e5R33-R38) we need to explicitly indicate the type.